app/controllers/verify/finance_controller.rb

```diff
@@ -3,20 +3,28 @@ class FinanceController < StepController
     helper_method :idv_finance_form
 
     def new
-      analytics.track_event(Analytics::IDV_FINANCE_VISIT)
+      analytics.track_event(Analytics::IDV_FINANCE_CCN_VISIT)
     end
 
     def create
       if idv_finance_form.submit(finance_params)
         idv_session.params = idv_finance_form.idv_params
         redirect_to verify_phone_url
       else
-        render :new
+        render_form
       end
     end
 
     private
 
+    def render_form
+      if finance_params[:finance_type] == 'ccn'
+        render :new
```

Comment:
Ah good catch, I have the plugin installed now too and should have noticed. Will add some coverage